### ISSUE TITLE
Fin<T> and Fails(): unexpected results

### ISSUE THREAD

**Issue by atruskie** [louthy/language-ext#1257]
I have a series of results, of varying types, each of which may be an error.

I deal with the error values elsewhere, but I wanted to accumulate all the errors at once. I did the following and expected it to work (written as a unit test):

```csharp
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using FluentAssertions;
    using LanguageExt;
    using LanguageExt.Common;

    [Fact]
    public void FinAndFails()
    {
      Error notAvailable = Error.New("not available");
    
      Fin<int> a = notAvailable;
      Fin<int> b = 3;
      Fin<string> c = notAvailable;
      Fin<string> d = "valid";
    
      var allErrors = a.Fails().Concat(b.Fails()).Concat(c.Fails()).Concat(d.Fails());
    
      allErrors.Should().BeEquivalentTo(new[] { notAvailable, notAvailable });
    }
```
Test output:

```
  Message: 
Expected allErrors to be a collection with 2 item(s), but found an empty collection.
```

The result was an empty collection. I expected a collection of the failures.

Now, I'm not sure what the correct behavior is:

On the one hand, it makes sense to me that `.Fails()` on a single Fin should work
  - It makes it symmetrical with a collection of Fins that a single Fin produces 1 error or 0 results
  - And if your `Fin<T>`s that you are trying to aggregate are of different types you can't do something like  `Seq.create(a, b, c, d).Fails()` because the generic parameters don't match

But it's also very clear that `.Fails()` is meant to work on a collection of Fins.

  https://github.com/louthy/language-ext/blob/bce0a161f8d07b0946b95c451daaeee8c732ee2a/LanguageExt.Core/Monads/Alternative%20Value%20Monads/Fin/Fin.Extensions.cs#L234-L235
  
In which case I'd argue that it is really confusing that it shows up as an essentially useless method on a singular Fin. Useless because no matter the state of the Fin, `.Fails()` on a single Fin will always yield an empty collection.

Using LanguageExt version 4.4.6

**Comment by aloslider** [comment]
Agree, is Fin's [`GetEnumerator`](https://github.com/louthy/language-ext/blob/bce0a161f8d07b0946b95c451daaeee8c732ee2a/LanguageExt.Core/Monads/Alternative%20Value%20Monads/Fin/Fin.cs#L387) supposed to yield on Success only?

**Comment by atruskie** [comment]
> Agree, is Fin's [`GetEnumerator`](https://github.com/louthy/language-ext/blob/bce0a161f8d07b0946b95c451daaeee8c732ee2a/LanguageExt.Core/Monads/Alternative%20Value%20Monads/Fin/Fin.cs#L387) supposed to yield on Success only?

I'd say yes. In the normal case when enumerating a single fin it should only yield if successful. That is how the other monads work.

The real issue is I think Fails() should be equivalent to a negative/bottom/inverse enumerator which has the opposite behavior of the normal GetEnumerator method.

If that's the case, then the fix here might be to add another overload to the Fails extension methods with the more specific `Fin<T>` parameter type.

**Comment by louthy** [comment]
Fixed in `v4.4.8`
